cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/connector/resource/AwsCloudWatchService.java

```diff
@@ -34,47 +32,46 @@
+    @Value("${aws.cloudwatch.threshold:1.0}")
     private double cloudwatchThreshhold;
 
-    @Value("${freeipa.aws.cloudwatch.max-batchsize:100}")
+    @Value("${aws.cloudwatch.max-batchsize:100}")
     private int maxBatchsize;
 
     @Inject
     private AwsClient awsClient;
 
-    public void addCloudWatchAlarmsForSystemFailures(List<CloudResource> instances, CloudStack stack, String regionName, AwsCredentialView credentialView) {
-        if (isCloudwatchEnabled(stack)) {
-            instances.stream().forEach(instance -> {
-                try {
-                    PutMetricAlarmRequest metricAlarmRequest = new PutMetricAlarmRequest();
-                    metricAlarmRequest.setAlarmActions(Arrays.asList("arn:aws:automate:" + regionName + ":ec2:recover"));
-                    metricAlarmRequest.setAlarmName(instance.getInstanceId() + alarmSuffix);
-                    metricAlarmRequest.setMetricName("StatusCheckFailed_System");
-                    metricAlarmRequest.setStatistic("Maximum");
-                    metricAlarmRequest.setNamespace("AWS/EC2");
-                    metricAlarmRequest.setDimensions(Arrays.asList(new Dimension().withName("InstanceId").withValue(instance.getInstanceId())));
-                    metricAlarmRequest.setPeriod(cloudwatchPeriod);
-                    metricAlarmRequest.setEvaluationPeriods(cloudwatchEvaluationPeriods);
-                    metricAlarmRequest.setThreshold(cloudwatchThreshhold);
-                    metricAlarmRequest.setComparisonOperator("GreaterThanOrEqualToThreshold");
-                    AmazonCloudWatchClient amazonCloudWatchClient = awsClient.createCloudWatchClient(credentialView, regionName);
-                    amazonCloudWatchClient.putMetricAlarm(metricAlarmRequest);
-                    LOGGER.debug("Created cloudwatch alarm for instanceId {}.", instance.getInstanceId());
-                } catch (AmazonCloudWatchException acwe) {
-                    LOGGER.error("Unable to create cloudwatch alarm for instanceId {}: {}", instance.getInstanceId(), acwe.getLocalizedMessage());
-                }
-            });
-        }
+    public void addCloudWatchAlarmsForSystemFailures(List<CloudResource> instances, String regionName, AwsCredentialView credentialView) {
+        AmazonCloudWatchClient amazonCloudWatchClient = awsClient.createCloudWatchClient(credentialView, regionName);
+
+        instances.stream().forEach(instance -> {
+            try {
+                PutMetricAlarmRequest metricAlarmRequest = new PutMetricAlarmRequest();
+                metricAlarmRequest.setAlarmActions(Arrays.asList("arn:aws:automate:" + regionName + ":ec2:recover"));
+                metricAlarmRequest.setAlarmName(instance.getInstanceId() + alarmSuffix);
+                metricAlarmRequest.setMetricName("StatusCheckFailed_System");
+                metricAlarmRequest.setStatistic("Maximum");
+                metricAlarmRequest.setNamespace("AWS/EC2");
```

Comment:
is it possible to create multiple cloudwatch alarms with one request to aws?

Reply:
This does not seem to be exposed in the API unfortunatelyt.